internal/services/gitserver/gitserver_test.go

```diff
@@ -0,0 +1,197 @@
+// Copyright 2019 Sorint.lab
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//     http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+package gitserver
+
+import (
+	"context"
+	"errors"
+	"io/ioutil"
+	"os"
+	"path/filepath"
+	"testing"
+	"time"
+
+	"agola.io/agola/internal/services/config"
+	"agola.io/agola/internal/util"
+)
+
+const (
+	commitOldTime = "2015-01-15 23:40:47 +0100 CET"
+	branchName    = "master"
+	tagName       = "v1.0"
+)
+
+func createTag(t *testing.T, git *util.Git, commitDate string) {
+	if _, err := git.Output(context.Background(), nil, "branch", "test"); err != nil {
```

Comment:
I'll change `commitDate string` to `commitTime time.Time` and convert it to a string inside this function.